### PR TITLE
[flink] Implements FlinkCatalog#getPartition

### PR DESCRIPTION
**Pull request by reswqa**
### Purpose

Implements `FlinkCatalog#getPartition`.

### Tests

IT

### API and Format

No.

### Documentation

No.

### CODE REVIEW

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkCatalog.java [hunk 1]
```diff
@@ -126,12 +128,16 @@
...
 import static org.apache.paimon.flink.utils.FlinkCatalogPropertiesUtil.serializeNewWatermarkSpec;
 import static org.apache.paimon.utils.Preconditions.checkArgument;
+import static org.apache.paimon.utils.Preconditions.checkNotNull;
 
 /** Catalog for paimon. */
 public class FlinkCatalog extends AbstractCatalog {
 
     private static final Logger LOG = LoggerFactory.getLogger(FlinkCatalog.class);
-
+    public static final String NUM_ROWS_KEY = "partition.numRows";
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkCatalog.java:137]: I suggest establishing a standard under the definition of the Flink community.

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkCatalog.java [hunk 2]
```diff
@@ -922,7 +928,34 @@ public final List<CatalogPartitionSpec> listPartitionsByFilter(
     @Override
     public CatalogPartition getPartition(ObjectPath tablePath, CatalogPartitionSpec partitionSpec)
             throws PartitionNotExistException, CatalogException {
-        throw new PartitionNotExistException(getName(), tablePath, partitionSpec);
+        checkNotNull(partitionSpec, "partition spec shouldn't be null");
+        Identifier identifier = toIdentifier(tablePath);
+        try {
+            Table table = catalog.getTable(identifier);
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkCatalog.java:934]: Can you just extract a method for reusing codes from `getPartitionSpecs`?

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkCatalog.java [hunk 3]
```diff
@@ -126,12 +127,16 @@
...
 
 /** Catalog for paimon. */
 public class FlinkCatalog extends AbstractCatalog {
 
     private static final Logger LOG = LoggerFactory.getLogger(FlinkCatalog.class);
-
+    public static final String NUM_ROWS_KEY = "numRows";
+    public static final String LAST_UPDATE_TIME_KEY = "lastUpdateTime";
+    public static final String FILE_SIZE_IN_BYTES_KEY = "fileSizeInBytes";
+    public static final String FILE_COUNT_KEY = "fileCount";
```
Comment (lsyldliu) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkCatalog.java:139]: numFiles? Hive name it.

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkCatalog.java [hunk 4]
```diff
@@ -126,12 +127,16 @@
...
+import static org.apache.paimon.utils.Preconditions.checkNotNull;
 
 /** Catalog for paimon. */
 public class FlinkCatalog extends AbstractCatalog {
 
     private static final Logger LOG = LoggerFactory.getLogger(FlinkCatalog.class);
-
+    public static final String NUM_ROWS_KEY = "numRows";
+    public static final String LAST_UPDATE_TIME_KEY = "lastUpdateTime";
+    public static final String FILE_SIZE_IN_BYTES_KEY = "fileSizeInBytes";
```
Comment (lsyldliu) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkCatalog.java:138]: Hive is `totalSize`, does this options is represent the same means of hive?

**Review by lsyldliu — COMMENTED** [review]
Thanks for your contribution, I left some comments.

**Review by lsyldliu — APPROVED** [review]
LGTM

**Review by JingsongLi — APPROVED** [review]
+1

**Comment by JingsongLi** [comment]
@reswqa Is there any SQL will invoke this method?

**Comment by JingsongLi** [comment]
> @reswqa Is there any SQL will invoke this method?

Currently, no..

**Pull request merged**